### PR TITLE
CChunkData optimizations

### PR DESCRIPTION
**Pull request by Alvsch**
Optimized CChunkData::write
Now creating rayon threads before the tokio runtime
Added profiling profile

### CODE REVIEW

File: pumpkin-protocol/src/client/play/c_chunk_data.rs [hunk 1]
```diff
@@ -124,22 +146,6 @@ impl<'a> ClientPacket for CChunkData<'a> {
...
-                    continue;
-                }
-                let index = i / 2;
-                let mask = if i % 2 == 1 { 0xF0 } else { 0x0F };
-                chunk_light[index] |= mask;
-            }
-
-            lighting_subchunks.push(chunk_light);
-        });
-
```
Comment (Snowiiii) [inline pumpkin-protocol/src/client/play/c_chunk_data.rs:142]: Why did you remove light ?

Comment (Alvsch) [reply pumpkin-protocol/src/client/play/c_chunk_data.rs:142]: i moved it up into the other loop

**Comment by Snowiiii** [comment]
I wonder how you benchmarked the speed difference and what the speed difference is

**Comment by Alvsch** [comment]
I used both cargo-flamegraph and samply for profiling. (Both rust specific tools)

The total per chunk loading on my pc was reduced from ~40 ms to ~20 ms
This being flying around with 32 render distance.

**Comment by Snowiiii** [comment]
Sounds very cool!
Thanks @Alvsch
